### PR TITLE
Add Gaz2 requested query

### PR DESCRIPTION
**Pull request by ICoxCH**

### CODE REVIEW

File: src/main/java/uk/gov/ch/repository/transaction/Gaz2TransactionRepository.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.data.repository.PagingAndSortingRepository;
+import uk.gov.ch.model.transaction.sqldatamodels.Gaz2TransactionDataModel;
+
+public interface Gaz2TransactionRepository extends PagingAndSortingRepository<Gaz2TransactionDataModel, Long> {
+    @Query(value = "select tr.TRANSACTION_ID, tr.TRANSACTION_TYPE_ID, tr.TRANSACTION_STATUS_TYPE_ID " +
+            "from corporate_body cb, transaction tr " +
+            "where tr.CORPORATE_BODY_ID = cb.CORPORATE_BODY_ID " +
+            "and tr.TRANSACTION_TYPE_ID in (541, 542) " +
+            "and tr.TRANSACTION_STATUS_TYPE_ID = 26 " +
+            "and cb.INCORPORATION_NUMBER = ?1",
```
Comment (markpit) [inline src/main/java/uk/gov/ch/repository/transaction/Gaz2TransactionRepository.java:13]: I would go with a named param here, easier to read and less error prone - https://www.baeldung.com/spring-data-jpa-query#named-parameters

Comment (markpit) [reply src/main/java/uk/gov/ch/repository/transaction/Gaz2TransactionRepository.java:13]: could there be a resultSet returned with size > 1? 
If a company has been dissolved several times, what does it's transaction history look like? I thought each time a change was made to a company a new transaction was created which might mean there could be > 1 gaz2 with status of 26. Might be worth checking with a dba (mikee)

Comment (ICoxCH) [reply src/main/java/uk/gov/ch/repository/transaction/Gaz2TransactionRepository.java:13]: From Sarah:
"Connie/Dylan said you can have multiple issued but not requested."

So searching by requested should either return 0 or 1

**Comment by mattch1** [comment]
Just wondering why we go to the trouble or transforming what we get back from the query rather than just returning the data moel itself.

**Comment by ICoxCH** [comment]
> Just to understand this more clearly, I was wondering why we go to the trouble or transforming what we get back from the query rather than just returning the data model itself.

Over time the query/result from the query (entity class) could change depending on requirements.
By splitting into an entity class and a dto class. The query and entity class can be changed, but leaving the dto class the same, meaning that any services built against the existing dto will not suddenly stop working.

**Review by markpit — APPROVED** [review]

**Review by mattch1 — APPROVED** [review]

**Pull request merged**